### PR TITLE
Update documentation of BasicMolGANModel 

### PR DESCRIPTION
**Pull request by shreyasvinaya**
# Update documentation of BasicMolGANModel 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/molgan.py [hunk 1]
```diff
@@ -118,15 +118,21 @@ def create_generator(self) -> keras.Model:
         """
         Create generator model.
         Take noise data as an input and processes it through number of
-        dense and dropout layers. Then data is converted into two forms
-        one used for training and other for generation of compounds.
+        dense and dropout layers.
```
Comment (rbharath) [inline deepchem/models/molgan.py:121]: Can you remove these formatting changes?

File: deepchem/models/molgan.py [hunk 2]
```diff
@@ -141,8 +147,10 @@ def create_discriminator(self) -> keras.Model:
             1. adjacency tensor, containing bond information
             2. nodes tensor, containing atom information
         The input vectors need to be in one-hot encoding format.
-        Use MolGAN featurizer for that purpose. It will be simplified
-        in the future release.
+        Use MolGAN featurizer for that purpose.
```
Comment (rbharath) [inline deepchem/models/molgan.py:150]: Can you remove the formatting changes?

File: deepchem/models/molgan.py [hunk 3]
```diff
@@ -118,15 +118,21 @@ def create_generator(self) -> keras.Model:
...
             1. edges
             2. nodes
-        The format differs depending on intended use (training or sample generation).
-        For sample generation use flag, sample_generation=True while calling generator
+        
+        The format differs depending on intended use (sample generation or training).
+        For sample generation, flag "sample_generation=True" to be used while calling generator
         i.e. gan.generators[0](noise_input, training=False, sample_generation=True).
-        In case of training, not flag is necessary.
+        For training, flag "sample_generation=True" is not necessary
```
Comment (rbharath) [inline deepchem/models/molgan.py:134]: This should be reworded. 

```
For training the model, set `sample_generation=False`
```

**Review by rbharath — COMMENTED** [review]
Thanks for the contribution! In general, better to avoid formatting changes in PRs modifying documentation. Can you remove those and update this PR?

File: deepchem/models/molgan.py (hunk 3 above)
Comment (shreyasvinaya) [reply deepchem/models/molgan.py]: Hi, i've made the suggested changes

File: deepchem/models/molgan.py [hunk 4]
```diff
@@ -122,10 +123,11 @@ def create_generator(self) -> keras.Model:
         The model has two outputs:
             1. edges
             2. nodes
-        The format differs depending on intended use (training or sample generation).
-        For sample generation use flag, sample_generation=True while calling generator
+        The format differs depending on intended use (sample generation or training).
+        For sample generation, flag "sample_generation=True" to be used while calling generator
```
Comment (rbharath) [inline deepchem/models/molgan.py:127]: The original version uses active voice which is better than passive voice. Can you revert back to active voice?

File: deepchem/models/molgan.py [hunk 5]
```diff
@@ -142,6 +144,7 @@ def create_discriminator(self) -> keras.Model:
         The input vectors need to be in one-hot encoding format.
         Use MolGAN featurizer for that purpose. It will be simplified
         in the future release.
+
```
Comment (rbharath) [inline deepchem/models/molgan.py:147]: Can you remove this empty line?

File: deepchem/models/molgan.py [hunk 6]
```diff
@@ -122,10 +123,11 @@ def create_generator(self) -> keras.Model:
...
             1. edges
             2. nodes
-        The format differs depending on intended use (training or sample generation).
-        For sample generation use flag, sample_generation=True while calling generator
+        The format differs depending on intended use (sample generation or training).
+        For sample generation, flag "sample_generation=True" to be used while calling generator
         i.e. gan.generators[0](noise_input, training=False, sample_generation=True).
-        In case of training, not flag is necessary.
+        For training the model, set `sample_generation=False`
+
```
Comment (rbharath) [inline deepchem/models/molgan.py:130]: Can you remove this empty line?

**Review by rbharath — COMMENTED** [review]
Looking better, but a few fixes still needed

**Comment by shreyasvinaya** [comment]
I have made the requested changes

**Comment by rbharath** [comment]
Changes look fine now. I've set the CI to run. If everything looks fine we can merge in

**Comment by shreyasvinaya** [comment]
Hi Bharath, 
can we merge the PR now?

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**